pkg/tfgen/pluginHost.go

```diff
@@ -36,6 +38,23 @@ func (p *inmemoryProvider) GetSchema(version int) ([]byte, error) {
 	return p.schema, nil
 }
 
+func (p *inmemoryProvider) GetPluginInfo() (workspace.PluginInfo, error) {
+	var version *semver.Version
+	if p.info.Version != "" {
+		v, err := semver.ParseTolerant(p.info.Version)
+		if err != nil {
+			return workspace.PluginInfo{}, fmt.Errorf("failed to parse pkg version: %w", err)
```

Comment:
Can we include the provider name in the error? I'm guessing that the version may be empty string at least some of the time, like if if it's malformed HCL e.g. `resource somenonexistentthing_someresource`.